> Confirmed, I can now build the project on NixOS with sandboxing enabled. ~Still I cannot run it,~ Binary runs, but fails. Is there some runtime dependencies I should have to test this?
> 
> ```shell
> $ ./result/bin/gspeech
> Traceback (most recent call last):
>   File "/nix/store/2521np3363z8vzk8358lb3mcpsb1pxka-gSpeech-0.9.1/bin/..gspeech-wrapped-wrapped", line 9, in <module>
>     sys.exit(main())
>   File "/nix/store/2521np3363z8vzk8358lb3mcpsb1pxka-gSpeech-0.9.1/lib/python3.7/site-packages/speech/main.py", line 249, in main
>     pid.kill_if_already_exist(conf.app_name, conf.pid)
>   File "/nix/store/2521np3363z8vzk8358lb3mcpsb1pxka-gSpeech-0.9.1/lib/python3.7/site-packages/speech/pid.py", line 23, in kill_if_already_exist
>     with open(pid_path, 'w') as f:
> FileNotFoundError: [Errno 2] No such file or directory: '/home/eeva/.cache/gSpeech/gspeech.pid'
> ```

--- a/pkgs/applications/audio/gspeech/default.nix
+++ b/pkgs/applications/audio/gspeech/default.nix
@@ -1,0 +1,73 @@
+{ lib
+, fetchFromGitHub
+, python3
+, gtk3
+, wrapGAppsHook
+, glibcLocales
+, gobject-introspection
+, gettext
+, pango
+, gdk-pixbuf
+, librsvg
+, atk
+, libnotify
+, libappindicator-gtk3
+, gst_all_1
+, makeWrapper
+, picotts
+, sox
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "gSpeech";
+  version = "0.9.2";
+
+  src = fetchFromGitHub {
+    owner = "mothsart";
+    repo = pname;
+    rev = version;
+    sha256 = "11pvdpb9jjssp8nmlj21gs7ncgfm89kw26mfc8c2x8w2q4h92ja3";
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook
+    gobject-introspection
+    pango
+    gdk-pixbuf
+    atk
+    gettext
+    libnotify
+    libappindicator-gtk3
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    makeWrapper
+  ];
+
+  buildInputs = [
+    glibcLocales
+    gtk3
+    python3
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pygobject3
+    librsvg
+  ];
+
+  postFixup = ''
+    wrapProgram $out/bin/gspeech --prefix PATH : ${lib.makeBinPath [ picotts ]}
+    wrapProgram $out/bin/gspeech-cli --prefix PATH : ${lib.makeBinPath [ picotts ]}
+  '';
+
+  strictDeps = false;
+
+  meta = with lib; {
+    description = "A minimal GUI for the Text To Speech 'Svox Pico'. Read clipboard or selected text in different languages and manage it : pause, stop, replay.";
+    homepage = "https://github.com/mothsART/gSpeech";
+    maintainers = with maintainers; [ mothsart ];
+    license = licenses.gpl3;
+    platforms = platforms.unix;
+  };
+}
+

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -19211,6 +19211,8 @@ in
 
   gpg-mdp = callPackage ../applications/misc/gpg-mdp { };
 
+  gspeech = callPackage ../applications/audio/gspeech { };
+
   icesl = callPackage ../applications/misc/icesl { };
 
   keepassx = callPackage ../applications/misc/keepassx { };

